backend/packages/Upgrade/src/api/services/ExperimentService.ts

```diff
@@ -213,7 +213,11 @@ export class ExperimentService {
 
   public async getSingleExperiment(id: string, logger?: UpgradeLogger): Promise<ExperimentDTO | undefined> {
     const experiment = await this.findOne(id, logger);
-    return this.reducedConditionPayload(this.formatingPayload(experiment));
+    if (experiment) {
+      return this.reducedConditionPayload(this.formatingPayload(experiment));
+    } else {
+      return null;
```

Comment:
You should return undefined here for the controller to throw appropriate error code. Event the function definition expects undefined.